site/src/util/workspace.ts

```diff
@@ -191,3 +191,25 @@ export const isWorkspaceOn = (workspace: Workspace): boolean => {
   const status = workspace.latest_build.job.status
   return transition === "start" && status === "succeeded"
 }
+
+export const workspaceQueryToFilter = (query?: string): WorkspaceFilter => {
+  let filter: WorkspaceFilter = {
+    Owner: "",
+    OrganizationID: "",
+  }
+
+  if (query) {
+    const parts = query.split(" ")
+
+    parts.map((part) => {
+      if (part.startsWith("owner:")) {
+        filter = {
+          Owner: part.slice("owner:".length),
+          OrganizationID: "",
+        }
+      }
+    })
+  }
+
+  return filter
+}
```

Comment:
This is _wonderful_. I have a few suggestions based on:

1. wanting to prepare the string
2. we as a FE try to never use `let`
3. we prefer explicit `if-else` structuring (opposite of Go, I know).

so here we are:

```suggestion
export const workspaceQueryToFilter = (query?: string): WorkspaceFilter => {
  const defaultFilter: WorkspaceFilter = {
    Owner: "",
    OrganizationID: "",
  }

  const preparedQuery = query?.replace(/  +/g, " ")

  if (!preparedQuery) {
    return defaultFilter
  } else {
    const parts = preparedQuery.split(" ")

    for (const part of parts) {
      const [key, val] = part.split(":")
      if (key === "owner") {
        return {
          Owner: val,
          OrganizationID: "",
        }
      }
    }

    return defaultFilter
  }
}
```

The `preparedQuery` just accounts for cases when more than one space is used. LMK if you have any questions!